ale/drivers/mex_drivers.py

```diff
@@ -322,77 +322,162 @@ def detector_center_sample(self):
+            num_records = self.label['FILE_RECORDS']
+            img_start_record = self.label['^IMAGE']
+            img_start_byte = bytes_per_record * (img_start_record - 1) # Offset by one for zero-based records
+            num_img_records = num_records - img_start_record
+            image_file.seek(img_start_byte)
+
+            for record in range(num_img_records):
+                record_bytes = image_file.read(bytes_per_record)
+                eph_time = struct.unpack('<d', record_bytes[:8])[0]
+                exp_dur = struct.unpack('<f', record_bytes[8:12])[0] / 1000
+                #if record == 0:
+                # Offset for zero-based corrections, and then offest for ISIS pixel definition
+                lines.append(record+1-0.5)
+                times.append(eph_time)
+                durations.append(exp_dur)
+            # Only add records if exposure duration has changed since the line before
                     #elif exp_dur != durations[-1]:
                     #    # Offset for zero-based corrections, and then offest for ISIS pixel definition
                     #    lines.append(record+1-0.5)
                     #    times.append(eph_time)
                     #    durations.append(exp_dur)
-            self._ephemeris_stop_time = times[-1] + durations[-1]
-            self._center_ephemeris_time = (self.ephemeris_stop_time + self.ephemeris_start_time) / 2
-            times = [time - self.center_ephemeris_time for time in times]
-            # print(self.ephemeris_stop_time, self.center_ephemeris_time)
+        self._binary_exposure_durations = durations
+        self._binary_lines = lines
+        self._binary_ephemeris_times = times
+
+
 
-            self._line_scan_rate = (lines, times, durations)
-        return self._line_scan_rate
     @property
     def center_ephemeris_time(self):
+        """
+        Returns the center ephemeris time.
+
+        For HRSC, the center ephemeris time is calculated with the ephemeris stop time,
+        which is calculated from the binary image data.
+
+        Returns
+        -------
+        : float
+          Center ephemeris time
+        """
         if not hasattr(self, '_center_ephemeris_time'):
-            self._center_ephemeris_time = (self.ephemeris_stop_time - self._ephemeris_start_time) / 2
+            self._center_ephemeris_time = (self.ephemeris_stop_time + self.ephemeris_start_time) / 2
         return self._center_ephemeris_time
 
+
     @property
     def ephemeris_stop_time(self):
+        """
+        Returns the ephemeris stop time.
+
+        For HRSC, the ephemeris stop time is calculated from the binary image data.
+
+        Returns
+        -------
+        : float
+          Ephemeris stop time
+        """
         if not hasattr(self, '_ephemeris_stop_time'):
-            _, _, _ = self.line_scan_rate
+            self._ephemeris_stop_time = self.binary_ephemeris_times[-1] + self.binary_exposure_durations[-1]
```

Comment:
Is there an entry in the binary lists for every line? Usually we reduce the line scan rate table down to only the first line of each new exposure duration.

Reply:
This logic was commented out in the code. I can put it back if that's what we want to do.

Reply:
Minimizing that table seems like a good idea. A full table would be 10,000's of records where a minimized table might be10's or less. I suggest putting it back in.